sweep/sweeper.go

```diff
@@ -880,13 +930,27 @@ func (s *UtxoSweeper) sweep(inputs inputSet,
 	return nil
 }
 
-// rescheduleInputs updates the pending inputs with the given tx inputs. It
-// increments the `publishAttempts` and calculates the next broadcast height
-// for each input. When the publishAttempts exceeds MaxSweepAttemps(10), this
-// input will be removed.
-func (s *UtxoSweeper) rescheduleInputs(inputs []*wire.TxIn) {
+// markInputsPendingPublish saves the sweeping tx to db and updates the pending
+// inputs with the given tx inputs. It increments the `publishAttempts` and
+// calculates the next broadcast height for each input. When the
+// publishAttempts exceeds MaxSweepAttemps(10), this input will be removed.
+//
+// TODO(yy): add unit test once done refactoring.
+func (s *UtxoSweeper) markInputsPendingPublish(tr *TxRecord,
+	tx *wire.MsgTx) error {
+
+	// Add tx to db before publication, so that we will always know that a
+	// spend by this tx is ours. Otherwise if the publish doesn't return,
+	// but did publish, we'd lose track of this tx. Even republication on
+	// startup doesn't prevent this, because that call returns a double
+	// spend error then and would also not add the hash to the store.
+	err := s.cfg.Store.StoreTx(tr)
+	if err != nil {
+		return fmt.Errorf("store tx: %w", err)
+	}
+
 	// Reschedule sweep.
-	for _, input := range inputs {
+	for _, input := range tx.TxIn {
 		pi, ok := s.pendingInputs[input.PreviousOutPoint]
```

Comment:
yeah this is very unfortunate, `pendingInput` is a struct type, and `pendingInputs` is an alias type, kinda all get mixed up here - so ended up with π, since it's a go-practice that we use abbreviations.

Reply:
ok these weird names are all fixed in the final PR!